foxsec_sqs_worker.py

```diff
@@ -0,0 +1,231 @@
+"""FoxSec SQS worker"""
+import os
+import datetime
+import functools
+import time
+import json
+import ipaddress
+import boto3
+from botocore.exceptions import ClientError
+from pprint import pprint
+
+# Config
+# region = os.environ.get('AWS_DEFAULT_REGION', 'us-east-1')
+WAF_IPSET_ID = os.environ.get('IPSET_ID')
+WAF_IPSET_EXPIRATION_HOURS = os.environ.get('IPSET_EXPIRATION_HOURS', 24)
+
+# Constants
+DYNAMODB = boto3.resource('dynamodb').Table('foxsec-waf')
+SQS = boto3.client('sqs')
+WAFREGIONAL = boto3.client('waf-regional')
+
+
+def parse_arn(arn):
+    """Parse ARN"""
+
+    elements = arn.split(':', 5)
+    result = {
+        'arn': elements[0],
+        'partition': elements[1],
+        'service': elements[2],
+        'region': elements[3],
+        'account': elements[4],
+        'resource': elements[5],
+        'resource_type': None
+    }
+
+    if '/' in result['resource']:
+        result['resource_type'], result['resource'] = result['resource'].split('/', 1)
+    elif ':' in result['resource']:
+        result['resource_type'], result['resource'] = result['resource'].split(':', 1)
+
+    return result
+
+
+def retry(retry_count=5, delay=5, allowed_exceptions=()):
+    """Decorator that allows function retries"""
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            for _ in range(retry_count):
+                try:
+                    result = func(*args, **kwargs)
+                    if result:
+                        return result
+                except allowed_exceptions as current_exception:
+                    last_exception = current_exception
+                # Instead of printing, consider using python logging module
+                print("%s: Waiting for %s seconds before retrying again"
```

Comment:
Use `.format()` instead of `%s`, both here and throughout the code. `%s` is deprecated.